### DISCUSSION TITLE
this is awesome! But....

### DISCUSSION THREAD

**Discussion by rycossette** [thygate/stable-diffusion-webui-depthmap-script#8] [General]
first of all, this is incredible. I'm a 3D artist, so being able to generate my own seamless textures with combined depth/displacement maps is a game changer. But I noticed that the way the depth is processed, on a tiled image doesn't seem quite right...

In this example I'd expect the dark areas on the color image to be represented with dark areas in the depth map. Since in this example you'd expect the height to be an average over the image, vs lower at the top and higher at the bottom. 

I don't know if this is just the way the model works that it's using, that part is beyond me. Just wanted to share.

<img width="1417" alt="image" src="https://user-images.githubusercontent.com/115579042/200054746-f697bc14-5ccc-4dae-82d1-1cf0d293d5c9.png">

**Comment by thygate** [comment]
Hi, 

Thanks for sharing !

I had not tested the seamless tiling with the depthmap generator myself, but indeed there are some scenarios where it fails, which is to be expected.

Another scenario where it (expectedly) fails, if you try some `m.c. escher` like constructions, the generated depthmaps are always physically correct, and thus don't go well with the actual picture that has an illusion of several depth layers interconnecting..

Try using a different model, dpt_large yields the highest quality results in my experience..

Hope you find good use for it.

Thanks.
